arch/arm64/boot/dts/qcom/sdm636-motorola-beckham.dts

```diff
@@ -0,0 +1,744 @@
+/dts-v1/;
+
+#include "sdm636.dtsi"
+#include "pm660.dtsi"
+#include "pm660l.dtsi"
+#include <dt-bindings/arm/qcom,ids.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/input/gpio-keys.h>
+#include <dt-bindings/leds/common.h>
+
+/delete-node/ &qhee_code;
+/delete-node/ &adsp_mem;
+/delete-node/ &qseecom_mem;
+/delete-node/ &rmtfs_mem;
+
+/ {
+	model = "Motorola Moto Z3 Play";
+	compatible = "motorola,beckham", "qcom,sdm636";
+	chassis-type = "handset";
+
+	qcom,msm-id = <QCOM_ID_SDM636 0x0>;
+	qcom,board-id = <0x4A 0xC000>;
+
+	battery: battery {
+		compatible = "simple-battery";
+
+		charge-full-design-microamp-hours = <3000000>;
+		voltage-min-design-microvolt = <3400000>;
+		voltage-max-design-microvolt = <4400000>;
+	};
+
+	chosen {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		stdout-path = "framebuffer0";
+
+		framebuffer0: framebuffer@9d400000 {
+			compatible = "simple-framebuffer";
+			reg = <0x0 0x9d400000 0x0 (1080 * 2160 * 4)>;
+			status = "okay";
+			width = <1080>;
+			height = <2160>;
+			stride = <(1080 * 4)>;
+			format = "a8r8g8b8";
+
+
+			/* Prevent unused clocks from being disabled; keep bootloader-enabled
+	 		 * display stack (and framebuffer) working.
+	 		 */
+			assigned-clocks = <&mmcc MDSS_MDP_CLK>, <&mmcc MDSS_VSYNC_CLK>; /* from mdp */
+			assigned-clock-rates = <300000000>, <19200000>;                 /* from mdp */
+			clocks = <&mmcc MDSS_AHB_CLK>, /* from mdss & mdp */
+				<&mmcc MDSS_AXI_CLK>,
+				<&mmcc MDSS_VSYNC_CLK>,
+				<&mmcc MDSS_MDP_CLK>,
+				<&mmcc BYTE0_CLK_SRC>, /* from mdss_dsi0 */
+				<&mmcc PCLK0_CLK_SRC>,
+				<&mmcc MDSS_BYTE0_CLK>,
+				<&mmcc MDSS_BYTE0_INTF_CLK>,
+				<&mmcc MNOC_AHB_CLK>,
+				<&mmcc MISC_AHB_CLK>,
+				<&mmcc MDSS_PCLK0_CLK>,
+				<&mmcc MDSS_ESC0_CLK>;
+			/* Prevent unused power domains from being disabled (similar to clocks) */
+			power-domains = <&rpmpd SDM660_VDDCX>, <&mmcc MDSS_GDSC>;
+
+			/* In order to allow simpledrm framebuffer to know
+			 * physical dimensions */
+			panel = <&fb_panel>;
+
+			fb_panel: fb-panel {
+				width-mm = <68>;
+				height-mm = <121>;
+			};
+		};
+	};
+
+	/*
+	 * Until we hook up type-c detection, we
+	 * have to stick with this. But it works.
+	 */
+	extcon_usb: extcon-usb {
+		compatible = "linux,extcon-usb-gpio";
+		id-gpios = <&tlmm 58 GPIO_ACTIVE_HIGH>;
+	};
+
+	gpio-keys {
+		status = "okay";
+		compatible = "gpio-keys";
+
+		key-volume-up {
+			label = "Volume Up";
+			gpios = <&pm660l_gpios 7 GPIO_ACTIVE_LOW>;
```

Comment:
Where is pinctrl state describing electrical properties of PM660L pin 7?
See PR #69